### PR TITLE
fallback to unittest when unittest2 is not available

### PR DESCRIPTION
**Pull request by pgajdos**
unittest2 is not neccessary in python3

### CODE REVIEW

**Comment by mcepl** [comment]
@pgajdos Well, the question remains: why is it needed for Python 2.7? (it would be nice to mention it in changelog/spec file)? Is it **really** needed?

**Comment by pgajdos** [comment]
@mcepl 
```
[    7s] ____ SerializableTransformTest.test_encode_with_custom_repr_returns_object _____
[    7s] 
[    7s] self = <rollbar.test.test_serializable_transform.SerializableTransformTest testMethod=test_encode_with_custom_repr_returns_object>
[    7s] 
[    7s]     def test_encode_with_custom_repr_returns_object(self):
[    7s]         class CustomRepr(object):
[    7s]             def __repr__(self):
[    7s]                 return {'hi': 'there'}
[    7s]     
[    7s]         start = {'hello': 'world', 'custom': CustomRepr()}
[    7s]     
[    7s]         serializable = SerializableTransform(whitelist_types=[CustomRepr])
[    7s]         result = transforms.transform(start, serializable)
[    7s] >       self.assertRegex(result['custom'], "<class '.*CustomRepr'>")
[    7s] E       AttributeError: 'SerializableTransformTest' object has no attribute 'assertRegex'
[    7s] 
```
I think it is lost time to deal with it more as 2.7 is eol.

**Comment by mcepl** [comment]
```py
import six

six.assertRegex(self, result['custom'], "<class '.*CustomRepr'>")
```

**Comment by pgajdos** [comment]
@mcepl 
> ```python
> import six
> 
> six.assertRegex(self, result['custom'], "<class '.*CustomRepr'>")
> ```

Unfortunately, that does not work for me so. 
 
> > I think it is lost time to deal with it more as 2.7 is eol.
> 
> Not for SLE (although `python-rollbar` is not in SLE).

I do not discuss downstream matters in upstream issues. (I do not see the issue with above patch in SLE, though, perhaps bit of reasoning would help me understand the problem.)

**Comment by letaniaferreira** [comment]
Looks good to me.

File: rollbar/test/__init__.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import unittest2
+try:
+    import unittest2 as unittest
```
Comment (mrunalk) [inline rollbar/test/__init__.py:2]: Thanks for your change. I don't think we even need to fallback on unittest2 as Python 2+ has been EOLed. Just move ahead with replacing it to unittest.

**Comment by pgajdos** [comment]
Done.

File: rollbar/test/__init__.py (hunk 1 above)
Comment (mcepl) [reply rollbar/test/__init__.py]: Well, it doesn’t save me. We have `rollbar` in Leap 15.* so we still build python2- packages. I don’t want to knowingly break python2 compatibility if I can help it. Do you know `rollbar` functionality available only in Python 3.* or unittest2 respectively?

Comment (mrunalk) [reply rollbar/test/__init__.py]: @mcepl, Thank you for your feedback. We are definitely not removing the support for python2 yet. If we can replace unittest2 with unittest without breaking existing tests and builds that will be great.

File: .travis.yml [hunk 2]
```diff
@@ -2,16 +2,6 @@ sudo: false
 language: python
 matrix:
   include:
-    - python: "2.7"
```
[the remark this answers is not in the document]
Comment (mcepl) [reply .travis.yml:5]: @pgajdos ???

**Comment by pgajdos** [comment]
Hmm, @mrunalk and @mcepl: so should I actually revert to (original) cdfc35b?

I do not insist on making this change myself, so feel free to do the change yourself, if you wish -- I can just file an issue, if you think it would be better.

**Comment by mcepl** [comment]
> Hmm, @mrunalk and @mcepl: so should I actually revert to (original) [cdfc35b](https://github.com/rollbar/pyrollbar/commit/cdfc35b32af7b83d2cc65d3837ff707828d1c965)?
> 
> I do not insist on making this change myself, so feel free to do the change yourself, if you wish -- I can just file an issue, if you think it would be better.

I still haven’t got a reply to my answer what capabilities of python3/unittest2 is used in the test suite. Do we need unittest2 (with Python 2.7) at all?

**Comment by pgajdos** [comment]
@mcepl 

See https://github.com/rollbar/pyrollbar/pull/340#issuecomment-649349929.

**Comment by mrunalk** [comment]
> Hmm, @mrunalk and @mcepl: so should I actually revert to (original) [cdfc35b](https://github.com/rollbar/pyrollbar/commit/cdfc35b32af7b83d2cc65d3837ff707828d1c965)?
> 
> I do not insist on making this change myself, so feel free to do the change yourself, if you wish -- I can just file an issue, if you think it would be better.

I am OK removing 'unittest2' but Not OK with removing support for python2 at this time. Alternatively I can merge your original patch of 'fallback to unittest when unittest2 is not available' if you can provide some explanation on what is motivating on this change that will help understand the context.

**Comment by mcepl** [comment]
> I am OK removing 'unittest2' but Not OK with removing support for python2 at this time.

I completely agree, see the linked pull request.

**Pull request closed without merging**